Fix UI gaps: data-backed logs/requests, worktime chart, safer nav, redirect, and input limits

ebarthur opened the pull request:
This PR addresses several UI and functional gaps:

* **Display real data** in **requests** and **app logs**, including:

  * Real-time graph updates
  * Filter support for dates and search queries

* **Implement worktime charts** in both dashboards to show active backend hours based on requests/logs.

* **Improve nav header UX**:

  * Shows friendly guidance when no app is selected
  * Prevents navigation to `App Logs` / `Requests` when `appId` is undefined

* **Add redirect logic** in `app.$app.tsx`:

  * Prevents error when route lacks loader by redirecting `/app/:app` to `/app`

* **Limit app name length** to 28 characters in the app form to maintain layout integrity

* **Fix minor typos**

ebarthur committed:
implement worktime chart

ebarthur committed:
refactor nav-header to handle app selection logic

-  this also handles cases where no app is selected and the 'requests|  app logs' button throws an error because param.app is undefined

ebarthur committed:
Add loader to app.$app.tsx to redirect users to /app

ebarthur committed:
update app logs and requests routes to show real data

ebarthur committed:
set app name limit to 28 chars

ebarthur committed:
fix typo and cleanup

ebarthur committed:
Merge branch 'art/ui-errors' into art/ui-gaps

ebarthur committed:
show app name length on add app form

blackmann commented on app/components/nav-header.tsx:
```diff
@@ -24,27 +29,37 @@ export function NavHeader() {
 						{
-							"!border-blue-500/70": isLogs,
-							"!border-rose-500/70": isRequests,
+							"!border-blue-500/70": isLogs && appCount > 0 && appId,
```
```suggestion
							"!border-blue-500/70": isLogs && appId,
```
Should be enough or?

ebarthur replied:
Correct

blackmann commented on app/components/nav-header.tsx:
```diff
@@ -24,27 +29,37 @@ export function NavHeader() {
-					>
-						App Logs
-					</Link>
+					{appCount > 0 && appId ? (
```
Here too

```suggestion
					{appId ? (
```

should be enough

ebarthur replied:
correct

blackmann commented on app/lib/requests-summary.ts:
```diff
@@ -0,0 +1,49 @@
+		formattedPeriod,
+		percentageChange,
+		isIncrease,
+		changeIcon,
+		changeColor,
```
This is for the UI to decide. Not this function's responsibility.

ebarthur replied:
Okay on it

blackmann commented on app/routes/app.$app.tsx:
```diff
@@ -1,6 +1,21 @@
+} from "@remix-run/node";
 import { prisma } from "~/lib/prisma.server";
 
+export const loader = async ({ request, params }: LoaderFunctionArgs) => {
+	const url = new URL(request.url);
+	const path = url.pathname.replace(/\/$/, "");
+
+	if (path === `/app/${params.app}`) {
+		return redirect("/app");
+	}
+
+	return null;
+};
+
```
I'm not sure I understand what's going on here

ebarthur replied:
It used to throw `'You made a GET request to "/app/$app/" but did not provide a `loader` for route "routes/app.$app", so there is no way to handle the request.'` if user manually hits `/app/$app` so I fixed it since it's not a route, we want users to see, there's nothing there.

blackmann replied:
Check which part of the code is calling it then and remove it. Maybe some hindsight

ebarthur replied:
The `/app/$app` route is not intended to be directly accessed i.e it's a server-only route used internally for handling actions. No part of the app explicitly calls it. I caught this by manually removing `/logs` or `/requests` from the URL, which exposed the unintended access. To prevent confusion or errors, I added a redirect from `/app/$app` to `/app`.

blackmann replied:
Then just redirect `/app/$app` to `/app/$app/requests`. One-liner

blackmann commented on package.json:
```diff
@@ -32,6 +32,7 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-hook-form": "^7.56.1",
+    "react-tooltip": "^5.28.1",
```
Let's remove this and use: https://floating-ui.com/docs/tooltip#reusable-tooltip-component

ebarthur replied:
alright consider it done

ebarthur committed:
resolve code nits

ebarthur commented:
Any specific plans on global error handling? Login attempt for instance throws a `badRequest()` for incorrect credentials? What's your reason for that?

blackmann commented:
> Any specific plans on global error handling? Login attempt for instance throws a `badRequest()` for incorrect credentials? What's your reason for that?

to handle that, change `throw` to `return` and handle that @ebarthur

ebarthur committed:
further address code nits:  UI improvements and login error responses and details

ebarthur committed:
save last accessed app for better UX

ebarthur committed:
fix cases where `requests` or `logs` may be part of the app slug, and ignore k6

ebarthur committed:
show actually logs as described on the header

ebarthur commented:
Summary of New Changes:

- Login UX: Users are now redirected to their last accessed app after logging in, using a persisted cookie.

- Log Fetching Logic: Refactored the `fetchLogs` function to correctly fetch logs within the intended date range (start → end) instead of extending up to Date.now().

- Header Button Styling: Fixed a bug where requests/logs in the app slug caused incorrect styling for the logs/app toggle buttons in the header.
 -  Minor UI Fixes: Applied several small visual tweaks for better consistency across views.

@blackmann

blackmann commented on app/routes/app.$app.requests.tsx:
```diff
@@ -25,6 +26,13 @@ export const loader = async ({ request, params }: LoaderFunctionArgs) => {
 
+	const { lastApp } = await getLastAppRedirect(request);
+
+	let currentApp: string | undefined;
```
Rename this `currentAppCookie` so that it's easy to know what it is down the code

ebarthur replied:
alright

ebarthur committed:
move redirect logic to app

blackmann commented on app/routes/app.$app.requests.tsx:
```diff
@@ -51,12 +75,39 @@ export const meta: MetaFunction = () => {
 			<RequestsGraph />
-			<Filter />
-			<RequestLogsTable />
+			<Filter onFilterChange={handleFilterChange} searchParams={searchParams} />
```
Since you're getting `searchParams` from the hook, then just use the hook in `Filter` instead of passing it
